packages/react-menu/src/components/MenuGroupHeader/useMenuGroupHeader.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps } from '@fluentui/react-utils';
+import { useMergedRefs } from '@fluentui/react-hooks';
+import { MenuGroupHeaderProps, MenuGroupHeaderState } from './MenuGroupHeader.types';
+import { useMenuGroupContext } from '../../menuGroupContext';
+
+const mergeProps = makeMergeProps<MenuGroupHeaderState>({});
+
+/**
+ * Given user props, returns state and render function for a MenuGroupHeader.
+ */
+export const useMenuGroupHeader = (
+  props: MenuGroupHeaderProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: MenuGroupHeaderProps,
+): MenuGroupHeaderState => {
+  const { headerId: id } = useMenuGroupContext();
+
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef<HTMLElement>(null)),
+      as: 'div',
+      id,
+    },
+    defaultProps,
+    resolveShorthandProps(props, []),
+  );
```

Comment:
I suggest to add a unit test that user can specify own `id`

```
<MenuGroup>
  <MenuGroupHeader id="foo" /> // "id" should be foo ✔️ 
</MenuGroup>
```

Reply:
Also added a similar test for `MenuGroup` for `aria-labelledby`